x/ccv/provider/keeper/consumer_lifecycle.go

```diff
@@ -353,36 +354,58 @@ func (k Keeper) MakeConsumerGenesis(
 	return gen, hash, nil
 }
 
-// BeginBlockStopConsumers iterates over the pending consumer proposals and stop/removes the chain if the stop time has passed
-func (k Keeper) BeginBlockStopConsumers(ctx sdk.Context) {
+// StopAndPrepareForConsumerRemoval sets the phase of the chain to stopped and prepares to get the state of the
+// chain removed after unbonding period elapses
+func (k Keeper) StopAndPrepareForConsumerRemoval(ctx sdk.Context, consumerId string) error {
+	// The phase of the chain is immediately set to stopped, albeit its state is removed later (see below).
+	// Setting the phase here helps in not considering this chain when we look at launched chains (e.g., in `QueueVSCPackets)
+	k.SetConsumerPhase(ctx, consumerId, types.ConsumerPhase_CONSUMER_PHASE_STOPPED)
+
+	// state of this chain is removed once UnbondingPeriod elapses
+	unbondingPeriod, err := k.stakingKeeper.UnbondingTime(ctx)
+	if err != nil {
+		return err
+	}
+	removalTime := ctx.BlockTime().Add(unbondingPeriod)
+
+	if err := k.SetConsumerRemovalTime(ctx, consumerId, removalTime); err != nil {
+		return errorsmod.Wrapf(types.ErrInvalidRemovalTime, "cannot set removal time: %s", err.Error())
```

Comment:
Let's use `ErrInvalid...` error message for invalid input. Here the removalTime is computed internally. 

```suggestion
		return fmt.Errorf("cannot set removal time (%s): %s", removalTime.String(), err.Error())
```